pkgs/applications/science/math/eukleides/default.nix

```diff
@@ -1,26 +1,43 @@
 { lib, stdenv, fetchurl, bison, flex, texinfo, readline, texLive }:
 
-let
-  name    = "eukleides";
+lib.fix (eukleides: stdenv.mkDerivation rec {
+  pname = "eukleides";
   version = "1.5.4";
-in
-stdenv.mkDerivation {
-  name = "${name}-${version}";
 
   src = fetchurl {
-    url = "http://www.eukleides.org/files/${name}-${version}.tar.bz2";
+    url = "http://www.eukleides.org/files/${pname}-${version}.tar.bz2";
     sha256 = "0s8cyh75hdj89v6kpm3z24i48yzpkr8qf0cwxbs9ijxj1i38ki0q";
   };
 
-  buildInputs = [bison flex texinfo readline texLive];
+  nativeBuildInputs = [ bison flex ];
 
-  preConfigure = "sed -i 's/ginstall-info/install-info/g' doc/Makefile";
-  installPhase = "mkdir -p $out/bin ; make PREFIX=$out install";
+  buildInputs = [ texinfo readline texLive ];
```

Comment:
Can you please investigate if the binaries look up texinfo and latex at runtime (i.e. they search for them in `PATH`, as opposed to hardcode the build time paths)?

Reply:
Looks like texinfo is needed only to build the manual so I moved that to `nativeBuildInputs`.
Latex is just invoked as `latex` in the shell scripts, so I would have to keep `texLive` in `buildInputs`, right? Or should the wrapper also change this?